cylc/uiserver/resolvers.py

```diff
@@ -15,10 +15,189 @@
+        >>> snake_to_kebab('')
+        ''
+        >>> snake_to_kebab(None)
+        Traceback (most recent call last):
+        TypeError: <class 'NoneType'>
+
+    """
+    if isinstance(snake, str):
+        if not snake:
+            return ''
+        return f'--{snake.replace("_", "-")}'
+    raise TypeError(type(snake))
+
+
+def check_cylc_version(version):
+    """Check the provided Cylc version is available on the CLI.
+
+    Sets CYLC_VERSION=version and tests the result of cylc --version
+    to make sure the requested version is installed and selectable via
+    the CYLC_VERSION environment variable.
+    """
+    proc = Popen(
+        ['cylc', '--version', '--color=never'],
+        env={**os.environ, 'CYLC_VERSION': version},
+        stdin=DEVNULL,
+        stdout=PIPE,
+        stderr=PIPE,
+        text=True
+    )
+    ret = proc.wait(timeout=5)
+    out, err = proc.communicate()
+    return ret or out.strip() == version
+
+
+class Services:
+    """Services provided by the UI Server."""
+
+    @staticmethod
+    async def play(workflows, args, workflows_mgr):
+        """Calls `cylc play`."""
+        response = []
+
+        # get ready to run the command
+        try:
+            # check that the requrest cylc version is available
+            cylc_version = None
+            if 'cylc_version' in args:
+                cylc_version = args['cylc_version']
+                if not check_cylc_version(cylc_version):
+                    return {'error': {'message': [
+                        f'cylc version not available: {cylc_version}'
+                    ]}}
+                args = dict(args)
+                args.pop('cylc_version')
+
+            # build the command
+            cmd = ['cylc', 'play', '--color=never']
+            for key, value in args.items():
+                if value is False:
+                    # don't add binary flags
+                    continue
+                key = snake_to_kebab(key)
+                if not isinstance(value, list):
+                    value = [value]
+                for item in value:
+                    cmd.append(quote(key))
+                    if item is not True:
+                        # don't provide values for binary flags
+                        if key == '--set':
+                            # TODO
+                            # don't quote this value as it messes with things
+                            # I think subprocess quotes everything anyway
+                            # double check
+                            cmd.append(item)
+                        else:
+                            cmd.append(quote(item))
+
+        except Exception as exc:
+            # oh noes, something went wrong, send back confirmation
+            response.append(
+                [
+                    {
+                        'error': {
+                            'message': str(exc)
+                        }
+                    }
+                ]
+            )
```

Comment:
Should we return the error response here instead of continuing the execution flow?

Reply:
Yeah, that would make sense!